lib/bundler/ui/shell.rb

```diff
@@ -10,8 +10,10 @@ class Shell
       attr_writer :shell
 
       def initialize(options = {})
-        if options["no-color"] || !$stdout.tty?
-          Thor::Base.shell = Thor::Shell::Basic
+        Thor::Base.shell = if options["no-color"] || !$stdout.tty?
+          Thor::Shell::Basic
+        else
+          Thor::Shell::Color
```

Comment:
We need to figure out what to do with this, since we have decided in https://github.com/bundler/bundler/pull/7002 that it's not the right fix. I'm going to give it some thought.